src/liquid/utils/e2e-tests.ts

```diff
@@ -83,3 +92,98 @@ export const getPageWithContent = async (
 
   return page
 }
+
+export const analyzeAccessibility = async (page, options = {}) => {
+  const defaultOptions = { rules: {} }
+  const disabledRuleIds = [
+    'document-title',
+    'html-has-lang',
+    'label',
+    'landmark-one-main',
+    'page-has-heading-one',
+    'region',
+  ]
+  disabledRuleIds.forEach((ruleId) => {
+    defaultOptions.rules[ruleId] = { enabled: false }
+  })
+
+  const finalOptions = Object.assign({}, defaultOptions, options)
+
+  // Inject the axe script in our page.
+  await page.addScriptTag({ path: resolvePath(PATH_TO_AXE) })
+  // Make sure that axe is executed in the next tick after
+  // the page emits the load event, giving priority to other scripts.
+  const accessibilityReport = await page.evaluate((axeOptions) => {
+    return new Promise((resolve) => {
+      setTimeout(resolve, 0)
+    }).then(() => axe.run(axeOptions))
+  }, finalOptions)
+
+  return accessibilityReport
+}
+
+function getInvalidNodeInfo(node) {
+  return `- ${printReceived(node.html)}\n\t${node.any
+    .map((check) => check.message)
+    .join('\n\t')}`
+}
+
+function getInvalidRuleInfo(rule) {
+  return `[rule id: ${rule.id}] ${printReceived(rule.help)} on ${
+    rule.nodes.length
+  } nodes\r\n${rule.nodes.map(getInvalidNodeInfo).join('\n')}`
+}
+
+declare global {
+  // eslint-disable-next-line @typescript-eslint/no-namespace
+  namespace jest {
+    interface Matchers<R> {
+      toHaveNoAccessibilityIssues(): R
+    }
+  }
+}
+
+// Add a new method to expect assertions with a very detailed error report
+expect.extend({
+  toHaveNoAccessibilityIssues(accessibilityReport, options) {
+    let violations = []
+    let incomplete = []
+
+    const defaultOptions = {
+      violationsThreshold: 0,
+      incompleteThreshold: 0,
+    }
+
+    const finalOptions = Object.assign({}, defaultOptions, options)
+
+    if (
+      accessibilityReport.violations.length > finalOptions.violationsThreshold
+    ) {
+      violations = [
+        `Expected to have no more than ${finalOptions.violationsThreshold} violations. Detected ${accessibilityReport.violations.length} violations:\n`,
+      ].concat(accessibilityReport.violations.map(getInvalidRuleInfo))
+    }
+
+    if (
+      finalOptions.incompleteThreshold !== false &&
+      accessibilityReport.incomplete.length > finalOptions.incompleteThreshold
+    ) {
+      incomplete = [
+        `Expected to have no more than ${finalOptions.incompleteThreshold} incomplete. Detected ${accessibilityReport.incomplete.length} incomplete:\n`,
+      ].concat(accessibilityReport.incomplete.map(getInvalidRuleInfo))
+    }
+
+    const message = [].concat(violations, incomplete).join('\n')
+    const pass =
+      accessibilityReport.violations.length <=
+        finalOptions.violationsThreshold &&
+      (finalOptions.incompleteThreshold === false ||
```

Comment:
In case you decide to pass `false` as option when executing the test.

> The incomplete array (also referred to as the "review items") indicates which nodes could neither be determined to definitively pass or definitively fail.